### PR TITLE
Add conversions

### PR DESCRIPTION
**Pull request by fivetran-avinash**
## PR Overview
**This PR will address the following Issue/Feature:** Add conversions project.

**This PR will result in the following new package version:** v0.9.0

Breaking change for users who were not already bringing in conversion fields via passthrough columns. This is designed to be backwards compatible for those cases.

**Please provide the finalized CHANGELOG entry which details the relevant changes included in this PR:**

# dbt_microsoft_ads v0.9.0
[PR #34](https://github.com/fivetran/dbt_microsoft_ads/pull/34) includes the following updates:

## Feature Updates: Conversion Support!
- We have added the following source fields to each `microsoft_ads` end model:
  - `conversions`: Number of conversions, measured by completion of an action by a customer after viewing your ad.
  - `conversions_value`: The revenue reported by the advertiser as a result of the `conversions` figure.
- In the event that you were already passing the above fields in via our [passthrough columns](https://github.com/fivetran/dbt_microsoft_ads?tab=readme-ov-file#passing-through-additional-metrics), the package will dynamically avoid "duplicate column" errors.
> The above new field additions are **breaking changes** for users who were not already bringing in conversion fields via passthrough columns.

## Under the Hood
- Created `microsoft_ads_persist_pass_through_columns` macro to ensure that the new conversion fields are backwards compatible with users who have already included them via passthrough fields.
- Added integrity and consistency validation tests within `integration_tests` folder for the transformation models (to be used by maintainers only).
- Updated seed files with new fields to test and validate on local data, provided specific casting to numeric (for currency) and integers. 

## Documentation Update
- Updated `microsoft_ads.yml` with new fields mentioned above. 

## Contributors
- [Seer Interactive](https://www.seerinteractive.com/?utm_campaign=Fivetran%20%7C%20Models&utm_source=Fivetran&utm_medium=Fivetran%20Documentation)

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt run –full-refresh && dbt test
- [NA] dbt run (if incremental models are present) && dbt test

Before marking this PR as "ready for review" the following have been applied:
- [NA] The appropriate issue has been linked, tagged, and properly assigned.
- [x] All necessary documentation and version upgrades have been applied.
    
- [x] docs were regenerated (unless this PR does not include any code or yml updates).
- [x] BuildKite integration tests are passing.
- [x] Detailed validation steps have been provided below.

### Detailed Validation
Please share any and all of your validation steps:

Using internal data we validated that
(1) data was passing through as expected
(2) customers leveraging passthrough columns were now only grabbing the necessary columns
Screenshot 2024-08-19 at 5 20 53 PM

Screenshot 2024-08-19 at 5 20 39 PM

Screenshot 2024-08-19 at 5 21 03 PM

(3) leveraged devprod on our internal data set to ensure validation tests were matching conversion values on all consistency and integrity tests

### If you had to summarize this PR in an emoji, which would it be?
 
🖱️

### CODE REVIEW

File: packages.yml [hunk 1]
```diff
@@ -1,3 +1,6 @@
 packages:
-- package: fivetran/microsoft_ads_source
-  version: [">=0.9.0", "<0.10.0"]
\ No newline at end of file
+  - git: https://github.com/fivetran/dbt_microsoft_ads_source.git
+    revision: feature/add-conversions
+    warn-unpinned: false
+  # - package: fivetran/microsoft_ads_source
```
Comment (fivetran-jamie) [inline packages.yml:5]: switch

**Review by fivetran-jamie — APPROVED** [review]

File: integration_tests/tests/consistency/consistency_account_report.sql [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    tags="fivetran_validations",
+    enabled=var('fivetran_validation_tests_enabled', false)
+) }}
+
+with prod as (
+    select
+        account_id,
+        sum(clicks) as clicks, 
+        sum(impressions) as impressions,
+        sum(spend) as spend
```
Comment (fivetran-joemarkiewicz) [inline integration_tests/tests/consistency/consistency_account_report.sql:11]: Not a requirement before release, but should we consider including the conversion fields in here as well (commented out) so we don't forget to include them in subsequent releases?

Comment (fivetran-joemarkiewicz) [reply integration_tests/tests/consistency/consistency_account_report.sql:11]: Same comment for other consistency tests

**Review by fivetran-joemarkiewicz — APPROVED** [review]
LGTM with one small suggestion in the tests to ensure we don't forget about conversions in future releases.

File: integration_tests/tests/consistency/consistency_account_report.sql (hunk 2 above)
Comment (fivetran-jamie) [reply integration_tests/tests/consistency/consistency_account_report.sql:11]: added

**Pull request merged**